src/webui/www/private/scripts/dynamicTable.js

```diff
@@ -1898,7 +1898,25 @@ window.qBittorrent.DynamicTable ??= (() => {
             this.newColumn("leeches", "", "QBT_TR(Leeches)QBT_TR[CONTEXT=TrackerListWidget]", 75, true);
             this.newColumn("downloaded", "", "QBT_TR(Times Downloaded)QBT_TR[CONTEXT=TrackerListWidget]", 100, true);
             this.newColumn("message", "", "QBT_TR(Message)QBT_TR[CONTEXT=TrackerListWidget]", 250, true);
+
+            this.initColumnsFunctions();
         },
+
+        initColumnsFunctions: function() {
+            const compareRows = function(row1, row2) {
+                if (!row1.full_data._sortable || !row2.full_data._sortable)
+                    return 0;
+
+                const value1 = this.getRowValue(row1);
+                const value2 = this.getRowValue(row2);
+                if ((typeof(value1) === "number") && (typeof(value2) === "number"))
+                    return compareNumbers(value1, value2);
```

Comment:
I would wish to avoid this check. Is it possible to determine the comparison function deterministically in `for (const column of this.columns)` loop?

Reply:
This is how the default comparing function works, I simply added first conditional to it. 

Maybe we could drop the type check altogether and only use compare() for columns in Trackers table? Some of them contain both numbers and strings (peers, seeds, leeches, times downloaded). IIRC numbers are coerced to strings and should be properly sorted. 

Alternatively, we could just create 3 different comparing functions and just assign them to appropriate columns (tier - numbers, url/status/msg - strings, and rest is mixed).

Or maybe we should do something else?

Reply:
>Alternatively, we could just create 3 different comparing functions and just assign them to appropriate columns (tier - numbers, url/status/msg - strings, and rest is mixed).

I would vote for this.

>Some of them contain both numbers and strings (peers, seeds, leeches, times downloaded, eg. when sorting by 'Times downloaded', one row can have 0 peers and another "N/A" peers). 

In GUI we would have an underlying representation for the `N/A` text, for example `-1`. So they would all have the same type when comparing.
In webui I don't think that is strictly necessary. We can allow conversions to string and if required, add `if` conditionals to tweak the preferred sorting/compare outcome.
This is just FYI.

Also AFAIK the js engine optimizer dislikes type uncertainty. So it implies that the type is preferred to be 'fixed' for most of the time and stay away from `typeof` conditionals as much as possible.